### PR TITLE
fix(argo-workflows): Accept multi auth mode for server

### PR DESCRIPTION
**Pull request by yu-croco**
Auth mode is supposed to accept multi values, so I fixed it.
Ref: https://github.com/argoproj/argo-helm/pull/2291#issuecomment-1798864481

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-workflows/templates/server/server-deployment.yaml [hunk 1]
```diff
@@ -52,8 +52,8 @@ spec:
           {{- with .Values.server.extraArgs }}
             {{- toYaml . | nindent 10 }}
           {{- end }}
-          {{- if .Values.server.authMode }}
-          - "--auth-mode={{ .Values.server.authMode }}"
+          {{- range .Values.server.authMode }}
+          - "--auth-mode={{ . }}"
```
Comment (yu-croco) [inline charts/argo-workflows/templates/server/server-deployment.yaml:56]: 📝 
ref: https://argoproj.github.io/argo-workflows/argo-server-auth-mode/
image

File: charts/argo-workflows/README.md [hunk 2]
```diff
@@ -260,7 +260,7 @@ Fields to note:
 | server.GKEmanagedCertificate.domains | list | `["argoworkflows.example.com"]` | Domains for the Google Managed Certificate |
 | server.GKEmanagedCertificate.enabled | bool | `false` | Enable ManagedCertificate custom resource for Google Kubernetes Engine. |
 | server.affinity | object | `{}` | Assign custom [affinity] rules |
-| server.authMode | string | `""` | Auth Mode is available from `server` , `client` or `sso`. If you chose `sso` , please configure `.Values.server.sso` as well. |
+| server.authMode | list | `[]` | Auth Mode is available from `server` , `client` or `sso`. If you chose `sso` , please configure `.Values.server.sso` as well. |
```
Comment (vladlosev) [inline charts/argo-workflows/README.md:263]: Maybe change it to `A list of supported authentication modes. Available values are server, client, or sso. If you provide sso, please configure .Values.server.sso as well.`. That makes it clearer that multiple values can be used.

Comment (yu-croco) [reply charts/argo-workflows/README.md]: Thank you for your review. I fixed in https://github.com/argoproj/argo-helm/pull/2336/commits/3366c7c98737398940e18467b4806f27b14f6a3e 🙋

File: charts/argo-workflows/values.yaml [hunk 3]
```diff
@@ -536,9 +536,9 @@ server:
     # - name: FOO
     #   value: "bar"
 
-  # -- Auth Mode is available from `server` , `client` or `sso`. If you chose `sso` , please configure `.Values.server.sso` as well.
+  # -- A list of supported authentication modes. Available values are `server`, `client`, or `sso`. If you provide sso, please configure `.Values.server.sso` as well.
   ## Ref: https://argoproj.github.io/argo-workflows/argo-server-auth-mode/
-  authMode: ""
+  authMode: []
```
Comment (jmeridth) [inline charts/argo-workflows/values.yaml:541]: Great change.  Thank you.

Could this break existing users who have it set to a string?  Should we add something in `NOTES.txt` about a deprecation and handle both for a bit?  I don't know if it is worth it. wdyt? 🤔

**Review by jmeridth — APPROVED** [review]

**Review by vladlosev — APPROVED** [review]
LGTM

**Pull request merged**